doc/kubernetes/modules/ROOT/pages/running/deployments/active-passive-sync.adoc

```diff
@@ -0,0 +1,214 @@
+Future versions of Keycloak and Infinispan plan to reduce those manual operations.
+
+* Degraded setups can lead to service or data loss if additional components fail.
+Monitoring is necessary to detect degraded setups.
+
+=== Failures this setup can survive
+
+[%autowidth]
+|===
+| Failure | Recovery | RPO^1^ | RTO^2^
+
+| Database node
+| If the writer instance fails, the database can promote a reader instance in the same or other datacenter to be the new writer.
+| No data loss
+| Seconds to minutes (depending on the database)
+
+| Keycloak node
+| Multiple Keycloak instances run in each datacenter. If one instance fails, it takes a few seconds for the other nodes to notice the change, and some incoming requests might receive an error message or are delayed for some seconds.
+| No data loss
+| Less than one minute
+
+| Infinispan node
+| Multiple Infinispan instances run in each datacenter. If one instance fails, it takes a few seconds to the other nodes to notice the change. Sessions are stored in at least two Infinispan nodes, so a single node failure doesn't lead to data loss.
+| No data loss
+| Less than one minute
+
+| Connectivity Infinispan
+| If the connectivity between the two datacenters is lost, session information can't be sent to the other datacenter.
+Incoming requests might receive an error message or are delayed for some seconds.
+The primary site marks the secondary site offline, and will stop sending data to the secondary.
+The setup is degraded until the connection is restored and the session data is re-synchronized to the secondary datacenter.
+| No data loss ^3^
+| Less than one minute
+
+| Connectivity Database
+| If the connectivity between the two datacenters is lost, the synchronous replication will fail, and it might take some time for the primary site to mark the secondary offline.
+Some requests might receive an error message or are delayed for some seconds.
+Manual operations might be necessary depending on the database.
+| No data loss ^3^
+| Seconds to minutes (depending on the database)
+
+| Primary Datacenter
+| If none of the Keycloak nodes is available, the loadbalancer will detect the outage and redirect the traffic to the secondary site.
+Some requests might receive an error message while the loadbalancer hasn't detected the primary datacenter failure.
+The setup will be degraded until the primary site is back up and the session state has been manually synced from the secondary to the primary site.
+| No data loss^3^
+| Less than one minute
+
+| Secondary Datacenter
+| If the secondary datacenter is not available, it will take a moment for the primary Infinispan and database to mark the secondary datacenter offline.
+Some requests might receive an error message while the detection takes place.
+Once the secondary datacenter is up again, the session state needs to be manually synced from the primary site to the secondary site.
+| No data loss^3^
+| Less than one minute
+
+|===
```

Comment:
It mentions connectivity between datacenters but I think it is missing the connectivity issue in a datacenter: keycloak -> database and keycloak -> infinispan.
Those should not incur any data loss and should switch to the secondary site(?)

Reply:
At the moment we're advocating to use the `/health/live` probe for the loadbalancer to decide with site to use. As that probe doesn't fail when either the database or the external Infinspan fail. 

If we would switch to `/health/ready`, we would check that the database is available. At the same time this would be a blocking probe, and it would fail in overload situations, which we don't want. At the moment there is no check or probe on the external Infinispan. I'll raise a new issue for that.  

I'll update the docs to include a statement about Infinispan and Keycloak failure.

Reply:
Update: If the writer Database instance fails, Aurora will promote a reader to be the new writer, and Keycloak will switch to the new writer after a small delay. This is covered in the first row of the table.

Reply:
Added a snippet about the Infinispan failure. It is actually quite disastrous. Let's see if we can improve on this in another issue.

```
| Infinispan cluster failure
| If the Infinispan cluster fails in the active datacenter, Keycloak won't be able to send session data to the secondary datacenter.
The state between Keycloak and Infinispan is out-of-sync, and also the state between the two Datacenters.
Keycloak will continue on the best effort basis, but the service might be degraded due to retry mechanisms.
Even when the Infinispan cluster is restored, its data will be out-of-sync with Keycloak.

Manual switchover to the secondary datacenter is recommended. As that datacenter is out-of-sync, the customer should consider if all data should be cleared from the session store of the passive datacenter to avoid out-of-date information.
| Loss of service and data
| Human intervention required
```

Reply:
The error is propagated to the user so Keycloak wouldn't make any progress. For sure you have a loss of service, but isn't it detected by the LB probes? Wouldn't the switch to secondary happen automatically?

Reply:
As discussed, Keycloak will be unavailable, which prevents the two sites to diverge. This make this simpler. Human intervention is still required to re-sync. Will update the docs in the next iteration.